src/avram/charms/enum_extensions.cr

```diff
@@ -28,10 +28,37 @@ abstract struct Enum
       end
     end
 
+    def parse(value : Array(T))
+      SuccessfulCast(Array(T)).new value
+    end
+
+    def parse(values : Array(Int))
+      results = values.map { |i| parse(i) }
+      if results.all?(SuccessfulCast)
+        parse(results.map(&.value.as(T)))
+      else
+        FailedCast.new
+      end
+    end
+
+    # def parse(values : Array(String))
+    #   results = values.map {|s| parse(s)}
+    #   if results.all?(SuccessfulCast)
+    #     parse(results.map(&.value.as(T)))
+    #   else
+    #     FailedCast.new
+    #   end
+    # end
```

Comment:
oops. Forgot to remove :joy:  This is handled in `Type`, so it's not needed here.